server/data_tiledb/tiledb_engine.py

```diff
@@ -0,0 +1,288 @@
+import os
+import json
+from server.data_common.driver import CXGDriver
+import tiledb
+import numpy as np
+import pandas as pd
+from server.data_common.fbs.matrix import encode_matrix_fbs
+from server.common.utils import path_join
+from server_timing import Timing as ServerTiming
+from server.common.constants import Axis
+
+
+class TileDbEngine(CXGDriver):
+    def __init__(self, location=None, config=None):
+        super().__init__(config)
+        self.url = location
+        self.update(location, config)
+        self.tiledb_ctx = tiledb.Ctx({
+            'sm.tile_cache_size': 8 * 1024 * 1024 * 1024,
+            'sm.num_reader_threads': 32,
+        })
+
+    def update(self, location=None, config=None):
+        self.config = config
+        if location:
+            self.url = location
+            if self.url[-1] != '/':
+                self.url += '/'
+
+            self._validate_and_initialize()
+
+    @staticmethod
+    def pre_checks(location):
+        if not TileDbEngine.isvalid(location):
+            raise RuntimeError(f"tiledb matrix is not valid: {location}")
+
+    @staticmethod
+    def file_size(location):
```

Comment:
I'm not sure returning zero is the right thing - shouldn't this return an error (or -1) indicating that size is unavailable?

Reply:
The only reason we have file_size is so the cli can print a warning that it might take a long time to load if the file size is too big.   Right now it only works with local h5ad files.   Perhaps the file_size should be removed and replaced with something else that more generally answers the question "is the dataset big enough to warn about", rather that file_size.